nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-db-schema-registry-service/src/test/java/org/apache/nifi/db/schemaregistry/DatabaseTableSchemaRegistryTest.java

```diff
@@ -0,0 +1,224 @@
+        runner.setProperty(dbcp, DATABASE_URL, url);
+        runner.setProperty(dbcp, DB_USER, String.class.getSimpleName());
+        runner.setProperty(dbcp, DB_PASSWORD, String.class.getName());
+        runner.setProperty(dbcp, DB_DRIVERNAME, "org.apache.derby.jdbc.EmbeddedDriver");
+        runner.enableControllerService(dbcp);
+    }
+
+    @Test
+    public void testGetSchemaExists() throws Exception {
+        DatabaseTableSchemaRegistry dbSchemaRegistry = new DatabaseTableSchemaRegistry();
+        runner.addControllerService("schemaRegistry", dbSchemaRegistry);
+        runner.setProperty(dbSchemaRegistry, DatabaseTableSchemaRegistry.DBCP_SERVICE, SERVICE_ID);
+        runner.setProperty(dbSchemaRegistry, DatabaseTableSchemaRegistry.SCHEMA_NAME, "SCHEMA1");
+        runner.enableControllerService(dbSchemaRegistry);
+        SchemaIdentifier schemaIdentifier = new StandardSchemaIdentifier.Builder()
+                .name("PERSONS")
+                .build();
+        RecordSchema recordSchema = dbSchemaRegistry.retrieveSchema(schemaIdentifier);
+        assertNotNull(recordSchema);
+        Optional<RecordField> recordField = recordSchema.getField("ID");
+        assertTrue(recordField.isPresent());
+        assertEquals(RecordFieldType.INT.getDataType(), recordField.get().getDataType());
+        recordField = recordSchema.getField("NAME");
+        assertTrue(recordField.isPresent());
+        assertEquals(RecordFieldType.STRING.getDataType(), recordField.get().getDataType());
+        recordField = recordSchema.getField("CODE");
+        assertTrue(recordField.isPresent());
+        assertEquals(RecordFieldType.INT.getDataType(), recordField.get().getDataType());
+        recordField = recordSchema.getField("DT");
+        assertTrue(recordField.isPresent());
+        assertEquals(RecordFieldType.DATE.getDataType(), recordField.get().getDataType());
+        // Get nonexistent field
+        recordField = recordSchema.getField("NOT_A_FIELD");
+        assertFalse(recordField.isPresent());
+    }
+
+    @Test
+    public void testGetSchemaNotExists() throws Exception {
+        DatabaseTableSchemaRegistry dbSchemaRegistry = new DatabaseTableSchemaRegistry();
+        runner.addControllerService("schemaRegistry", dbSchemaRegistry);
+        runner.setProperty(dbSchemaRegistry, DatabaseTableSchemaRegistry.DBCP_SERVICE, SERVICE_ID);
+        runner.setProperty(dbSchemaRegistry, DatabaseTableSchemaRegistry.SCHEMA_NAME, "SCHEMA1");
+        runner.enableControllerService(dbSchemaRegistry);
+        SchemaIdentifier schemaIdentifier = new StandardSchemaIdentifier.Builder()
+                .name("NOT_A_TABLE")
+                .build();
+        assertThrows(SchemaNotFoundException.class, () -> dbSchemaRegistry.retrieveSchema(schemaIdentifier));
+    }
+
+    private static class DBCPServiceSimpleImpl extends AbstractControllerService implements DBCPService {
```

Comment:
Recommend avoiding the use of `Impl` on class names, even for testing.
```suggestion
    private static class SimpleDBCPService extends AbstractControllerService implements DBCPService {
```